_posts/2025-05-05-Introducing-MCP-in-OpenSearch.md

```diff
@@ -0,0 +1,653 @@
+
+### Claude desktop Integration:
+
+Claude desktop natively supports MCP via stdio protocol, making integration straightforward:
+
+#### Step1: Configure the Claude Desktop
+
+* Go to Settings > Developer > Edit Config
+* Add the OpenSearch MCP server in the claude_desktop_config.json file
+
+```json
+{
+    "mcpServers": {
+        "opensearch-mcp-server": {
+            "command": "uvx",
+            "args": [
+                "test-opensearch-mcp"
+            ],
+            "env": {
+                // Required
+                "OPENSEARCH_URL": "<your_opensearch_domain_url>",
+
+                // For Basic Authentication
+                "OPENSEARCH_USERNAME": "<your_opensearch_domain_username>",
+                "OPENSEARCH_PASSWORD": "<your_opensearch_domain_password>",
+
+                // For IAM Role Authentication
+                "AWS_REGION": "<your_aws_region>",
+                "AWS_ACCESS_KEY": "<your_aws_access_key>",
+                "AWS_SECRET_ACCESS_KEY": "<your_aws_secret_access_key>",
+                "AWS_SESSION_TOKEN": "<your_aws_session_token>"
+            }
+        }
+    }
+}
+```
+
+#### Step 2: Chat away!
+
+Open a new chat in Claude desktop, and you'll see the available OpenSearch tools appear in your chat window. You can immediately start asking questions about your OpenSearch data.
+
+#### Demo
+![Claude Desktop Demo](/assets/media/blog-images/2025-05-05-Introducing-MCP-in-OpenSearch/claude-desktop-demo.png){: .img-fluid}
+
+### Conclusion
+
+The Standalone OpenSearch MCP Server bridges the gap between advanced AI capabilities and all versions of OpenSearch. By providing a version-agnostic solution, organizations can leverage the power of AI agents with their existing OpenSearch deployments without upgrading. Whether you're using Claude, LangChain, or other agent frameworks, the standalone server offers a standardized way to connect AI models with your valuable search data.
+
+Stay tuned for the official release, and we encourage you to test the demo version and provide feedback to help us improve this solution.
+
+# Section 2: OpenSearch MCP Client
+
+As part of our comprehensive MCP support, we're also adding MCP Client capabilities to the Agents within OpenSearch as an experimental feature. Starting with OpenSearch 3.0, the Conversational Agent and the newly introduced Plan, Execute, and Reflect agent can connect to external MCP servers and leverage their tools. Support for additional agent types will be available soon!
+
+For detailed implementation steps, see our documentation: https://github.com/opensearch-project/documentation-website/blob/ab15e100326a390251241e197fc9391deb09b095/_ml-commons-plugin/agents-tools/mcp/mcp-connector.md
+
+### Quick Start Guide:
```

Comment:
This part seems same with the Standalone MCP server readme . Should we ask user to refer to that readme ? I don't have strong opinion on this though.